mycore-user2/src/main/java/org/mycore/user2/MCRUserServlet.java

```diff
@@ -268,25 +324,20 @@ private void saveUser(HttpServletRequest req, HttpServletResponse res) throws Ex
             String pwd = u.getChildText("password");
             if ((pwd != null) && (pwd.trim().length() > 0) && user.getRealm().equals(MCRRealmFactory.getLocalRealm())) {
                 MCRUserManager.setUserPassword(user, pwd);
+                return user;
             }
         } else {
             user = MCRUserManager.getUser(userName, realmID);
             if (!(hasAdminPermission || currentUser.equals(user) || currentUser.equals(user.getOwner()))) {
                 res.sendError(HttpServletResponse.SC_FORBIDDEN);
-                return;
-            }
-        }
 
-        XPathExpression<Attribute> hintPath = XPathFactory.instance().compile("password/@hint", Filters.attribute());
-        Attribute hintAttr = hintPath.evaluateFirst(u);
-        String hint = hintAttr == null ? null : hintAttr.getValue();
-        if ((hint != null) && (hint.trim().length() == 0)) {
-            hint = null;
+            }
         }
-        user.setHint(hint);
-
-        updateBasicUserInfo(u, user);
+        return null;
```

Comment:
There needs to be a `return user;` after line 334, otherwise it will return null for a existing user and before line 334 there needs to be a `return null;`.